src/table.js

```diff
@@ -542,15 +550,81 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  switch (type) {
+    case "string":
+      return typeof value === "string" || value == null
+        ? value
+        : String(value);
+    case "boolean":
+      if (typeof value === "string") {
+        const trimValue = value.trim();
+        return trimValue === "true" ? true : trimValue === "false" ? false : null;
+      }
+      return typeof value === "boolean" || value == null
+        ? value
+        : Boolean(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : value === 0 || value === true || value === false
+        ? // eslint-disable-next-line no-undef
+          BigInt(value)
+        : !value || isNaN(value) || !Number.isInteger(+value)
+        ? NaN
+        : // eslint-disable-next-line no-undef
+          BigInt(value);
```

Comment:
I find the `value === 0` and `!value` tests difficult to follow given the myriad combinations of possible inputs. (For example, I’m thinking about empty strings, whitespace strings, objects that implement a valueOf method that returns zero or the empty string, etc.)

So, here is my attempt at rethinking it based on some of the ideas we discussed previously:

* Propagate bigints, null, and undefined as-is
* Otherwise, first coerce the input value to a number
* But, special-case falsey strings (empty/whitespace) to NaN rather than zero
* Then test if the number is an integer to avoid BigInt throwing an error
* Use undefined instead of NaN to represent missing data

```suggestion
      return typeof value === "bigint" || value == null
        ? value
        : Number.isInteger(typeof value === "string" && !value ? NaN : +value)
        ? BigInt(value) // eslint-disable-line no-undef
        : undefined;
```

The use of undefined is because NaN is a number type, not a BigInt type. So while it’s appropriate (even necessary to avoid a subsequent silent coercion of null to zero when using typed arrays) to use NaN to represent invalid data in the number case, I don’t think it applies here.

Reply:
Makes sense, thanks for thinking all that through and detailing it. Same for all this detailed feedback: much appreciated! 🙏